packages/middleware/test/build.test.ts

```diff
@@ -17,8 +25,30 @@ describe('build()', () => {
     );
     expect(middlewareManifest).toMatchSnapshot();
 
+    const outputFile = join(fixture, '.output/server/pages/_middleware.js');
+    expect(await fsp.stat(outputFile)).toBeTruthy();
+
+    require(outputFile);
+    //@ts-ignore
+    const middleware = global._ENTRIES['middleware_pages/_middleware'].default;
+    expect(typeof middleware).toStrictEqual('function');
+    const handledResponse = await middleware({
+      request: {
+        url: 'http://google.com',
+      },
+    });
+    const unhandledResponse = await middleware({
+      request: {
+        url: 'literallyanythingelse',
+      },
+    });
+    expect(String(handledResponse.response.body)).toEqual('Hi from the edge!');
+    expect(
+      (handledResponse.response as Response).headers.get('x-middleware-next')
+    ).toEqual(null);
+    expect(unhandledResponse.response.body).toEqual(null);
     expect(
-      await fsp.stat(join(fixture, '.output/server/pages/_middleware.js'))
-    ).toBeTruthy();
+      (unhandledResponse.response as Response).headers.get('x-middleware-next')
+    ).toEqual('1');
```

Comment:
@styfle Updated w/ a unit test that ensures returning nothing from userland middleware gets wrapped with a response w/ `x-middleware-next` set.